ArduPlane/commands_logic.cpp

```diff
@@ -720,10 +720,22 @@ bool Plane::verify_loiter_to_alt()
     //has target altitude been reached?
     if (condition_value != 0) {
         // primary goal, loiter alt
-        if (labs(condition_value - current_loc.alt) < 500 && loiter.sum_cd > 1) {
-            // primary goal completed, initialize secondary heading goal
-            condition_value = 0;
-            result = verify_loiter_heading(true);
+        if (labs(loiter.sum_cd) > 1) {
+            const bool reached_target_alt = labs(condition_value - current_loc.alt) < 500;
```

Comment:
well, I guess the .alt will be valid, just not the .lat/lng is garrenteed to be valid